hal/test-headers/rtc_api_tests.h

```diff
@@ -0,0 +1,97 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017-2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** \addtogroup hal_specification */
+/** @{*/
+
+#ifndef MBED_RTC_API_TESTS_H
+#define MBED_RTC_API_TESTS_H
+
+#include "device.h"
+
+#if DEVICE_RTC
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \defgroup ha_specificationl_rtc RTC
+ * RTC specification and tests
+ *
+ * To run the RTC hal tests use the command:
+ *
+ *     mbed test -t <toolchain> -m <target> -n tests-mbed_hal-rtc*
+ *
+ * # Defined behavior
+ * * The function ::rtc_init is safe to call repeatedly - Verified by test ::rtc_init_test
+ * * RTC accuracy is at least 10% - Not verified
+ * * Init/free doesn't stop RTC from counting - Verified by test ::rtc_persist_test
+ * * Software reset doesn't stop RTC from counting - Verified by ::rtc_reset_test
```

Comment:
All these constraints/assumptions should be documented in main header and porting guide.

Reply:
The API header has example code for each function and should be embedded in the porting guide. This test header serves as additional information that will also be in the porting guide. Keeping test declarations out of the API header makes it easier to understand since all the functions in that header need to be implemented by the porter.